src/zenml/cli/model_registry.py

```diff
@@ -0,0 +1,660 @@
+        except KeyError:
+            cli_utils.error(
+                f"Model with name {name} and version {version} does not exist."
+            )
+            return
+        cli_utils.pretty_print_model_version_details(model_version)
+
+    @models.command(
+        "delete-version",
+        help="Delete a model version from the active model registry.",
+    )
+    @click.argument(
+        "name",
+        type=click.STRING,
+        required=True,
+    )
+    @click.option(
+        "--version",
+        "-v",
+        type=str,
+        default=None,
+        help="Version of the model to delete.",
+        required=True,
+    )
+    @click.option(
+        "--yes",
+        "-y",
+        is_flag=True,
+        help="Don't ask for confirmation.",
+    )
+    @click.pass_obj
+    def delete_model_version(
+        model_registry: "BaseModelRegistry",
+        name: str,
+        version: str,
+        yes: bool = False,
+    ) -> None:
+        """Delete a model version from the active model registry.
+
+        Args:
+            model_registry: The model registry stack component.
+            name: Name of the model to delete.
+            version: Version of the model to delete.
+            yes: If set, don't ask for confirmation.
+        """
+        try:
+            model_registry.get_model_version(name, version)
+        except KeyError:
+            cli_utils.error(
+                f"Model with name {name} and version {version} does not exist."
+            )
+            return
+        if not yes:
+            confirmation = cli_utils.confirmation(
+                f"Found Model with the name `{name}` and the version `{version}`."
+                f"Do you want to delete it?"
+            )
+            if not confirmation:
+                cli_utils.declare("Model version deletion canceled.")
+                return
+        model_registry.delete_model_version(name, version)
+        cli_utils.declare(
+            f"Model {name} version {version} deleted successfully."
+        )
+
+    @models.command(
+        "update-version",
+        help="Update a model version in the active model registry.",
+    )
+    @click.argument(
+        "name",
+        type=click.STRING,
+        required=True,
+    )
+    @click.option(
+        "--version",
+        "-v",
+        type=str,
+        default=None,
+        help="Version of the model to update.",
+        required=True,
+    )
+    @click.option(
+        "--description",
+        "-d",
+        type=str,
+        default=None,
+        help="Description of the model to update.",
+    )
+    @click.option(
+        "--metadata",
+        "-m",
+        type=(str, str),
+        default=None,
+        help="Metadata or Tags to add to the model. can be used like: --m key1 value1 -m key2 value",
+        multiple=True,
+    )
+    @click.option(
+        "--stage",
+        "-s",
+        type=click.Choice(["None", "Staging", "Production", "Archived"]),
```

Comment:
No other possible choices people might want to have for stages custom to their particular process?

Reply:
This is not my code, just renamed the file to make more sense and free up `model.py` available 🙂